frankenstein/distributor.go

```diff
@@ -169,21 +169,29 @@ func (d *Distributor) Append(ctx context.Context, samples []*model.Sample) error
 		samplesByIngester[collector.Hostname] = append(otherSamples, sample)
 	}
 
-	// TODO paralellise
+	errs := make(chan error)
 	for hostname, samples := range samplesByIngester {
-		client, err := d.getClientFor(hostname)
-		if err != nil {
-			return err
-		}
-		err = timeRequestHistogramStatus(d.sendDuration, nil, func() error {
-			return client.Append(ctx, samples)
-		})
-		if err != nil {
-			return err
+		go func(ctx context.Context, hostname string, samples []*model.Sample) {
+			errs <- d.sendSamples(ctx, hostname, samples)
+		}(ctx, hostname, samples)
```

Comment:
`samples` changes on each iteration so has to be passed (IIUC). `ctx` can be closed over though. Updated accordingly.